components/nexus-security/src/main/java/org/sonatype/security/realms/privileges/AbstractPrivilegeDescriptor.java

```diff
@@ -34,6 +39,13 @@ public void installDependencies(final ConfigurationIdGenerator idGenerator) {
     this.idGenerator = idGenerator;
   }
 
+  @Override
+  public Permission createPermission(final CPrivilege privilege) {
+    assert privilege != null;
+    assert getType().equals(privilege.getType());
```

Comment:
FTR this is also assert vs. precondition since the only caller of this verifies this already.